File: Python/4MedianofTwoSortedArrays.py
__author__ = 'Minyi'
class Solution(object):
    def findMedianSortedArrays(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: float
        """
        n = len(nums1) + len(nums2)
        if n == 0:
            return 0

        merge = []
        while len(nums1)>0 and len(nums2)>0:
            if(nums1[0]<nums2[0]):
                merge.append(nums1[0])
                del nums1[0]
            else:
                merge.append(nums2[0])
                del nums2[0]
        merge.extend(nums1)
        merge.extend(nums2)

        if n%2!=0 :
            return merge[n//2]
        else:
            return (merge[n/2]+merge[n/2-1])/2.0

File: CPP/4MedianofTwoSortedArrays.cpp
class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
        vector<int> merge;
        if(nums1.size()==0 && nums2.size()==0){
        	return 0;
        }
		if(nums1.size()==0){
			for(int i=0; i<nums2.size(); i++){
				merge.push_back(nums2[i]);
			}
		}else if(nums2.size()==0){
			for(int i=0; i<nums1.size(); i++){
				merge.push_back(nums1[i]);
			}
		}else{
			if(nums1[0]>=nums2[nums2.size()-1]){
				for(int i=0; i<nums2.size(); i++){
					merge.push_back(nums2[i]);
				}
				for(int i=0; i<nums1.size(); i++){
					merge.push_back(nums1[i]);
				}
			}else if(nums2[0]>=nums1[nums1.size()-1]){
				for(int i=0; i<nums1.size(); i++){
					merge.push_back(nums1[i]);
				}
				for(int i=0; i<nums2.size(); i++){
					merge.push_back(nums2[i]);
				}
			}else{
				int i=0, j=0;
				while(i<nums1.size() && j<nums2.size()){
					if(nums1[i]<nums2[j]){
						merge.push_back(nums1[i]);
						i++;
					}else{
						merge.push_back(nums2[j]);
						j++;
					}	
				}
				
				if(i<nums1.size()){
					for(;i<nums1.size();i++){
						merge.push_back(nums1[i]);
					}
				}
				
				if(j<nums2.size()){
					for(;j<nums2.size(); j++){
						merge.push_back(nums2[j]);
					}
				}
			}
		}
		if(merge.size()%2!=0){
			return merge[merge.size()/2];
		}else{
			return ( merge[merge.size()/2] + merge[merge.size()/2-1]) / 2.0;
		}
    }
};
